cpp/dolfinx/geometry/utils.h

```diff
@@ -791,20 +786,30 @@ determine_point_ownership(const mesh::Mesh<T>& mesh, std::span<const T> points)
     std::swap(recv_offsets, send_offsets);
   }
 
-  std::vector<std::int32_t> recv_ranks(recv_offsets.back());
+  // Get distances from closest entity of points that were on the other process
+  std::vector<T> recv_distances(recv_offsets.back());
   MPI_Neighbor_alltoallv(
-      cell_indicator.data(), send_sizes.data(), send_offsets.data(),
-      dolfinx::MPI::mpi_type<std::int32_t>(), recv_ranks.data(),
-      recv_sizes.data(), recv_offsets.data(),
-      dolfinx::MPI::mpi_type<std::int32_t>(), reverse_comm);
+      squared_distances.data(), send_sizes.data(), send_offsets.data(),
+      dolfinx::MPI::mpi_type<T>(), recv_distances.data(), recv_sizes.data(),
+      recv_offsets.data(), dolfinx::MPI::mpi_type<T>(), reverse_comm);
 
   std::vector<std::int32_t> point_owners(points.size() / 3, -1);
-  for (std::size_t i = 0; i < unpack_map.size(); i++)
+  std::vector<T> closest_distance(points.size() / 3, -1);
+  std::stringstream qq;
```

Comment:
?

Reply:
Stringstream is remaining from debugging. Will remove